mlflow/metrics/utils/make_genai_metric.py

```diff
@@ -0,0 +1,206 @@
+from typing import Any, Dict, List, Union
+
+from mlflow.exceptions import MlflowException
+from mlflow.metrics.base import EvaluationExample, MetricValue
+from mlflow.models import EvaluationMetric, make_metric
+from mlflow.protos.databricks_pb2 import INTERNAL_ERROR, INVALID_PARAMETER_VALUE
+from mlflow.pyfunc import PyFuncModel, _load_model_or_server
+from mlflow.utils import env_manager as _EnvManager
+from mlflow.utils.class_utils import _get_class_from_string
+
+
+def make_genai_metric(
+    name: str,
+    version: str,
```

Comment:
Do we need to enable users to version metrics themselves, or is it sufficient for us to version built-in metrics? IMO, version should at least be optional for make_genai_metric